js/userscript.js

```diff
@@ -1,51 +1,253 @@
+  let favShowsHeading = document.getElementById('favShowsHeading');
+  favShowsHeading.innerText = 'Favourite Shows';
+  let userDetails = JSON.parse(localStorage.getItem('users'));
+
+  let signedInUser = userDetails.filter((user) => user.signedIn === true);
+  console.log(signedInUser);
+
+
+  if (!signedInUser[0].favShows) {
+    let noFavShowText = `<h4>Click the TV icon to search for and add your favourite shows</h4>`;
+    return usersPage.innerHTML = noFavShowText;
+  }
+  let favShows = signedInUser[0].favShows;
+  let shows = getAllShows();
+  let favShowsList = shows.filter((show) => {
+    for (let i = 0; i < favShows.length; i++) {
+      if (show.id == favShows[i]) {
+        return show;
+      }
+    }
+  });
+  let favShowsHtml = favShowsList.map((show) => {
+    let showImg = show.image?.medium ?? `https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRqEWgS0uxxEYJ0PsOb2OgwyWvC0Gjp8NUdPw&usqp=CAU`;
+    return `
+        <div class="episode-card card text-center col-2 col-md-3">
+          <div class="card-header">
+            <h2 class="card-header-text ">${show.name} <span class="user-fav-star-rating"><i class="user-fav-star-rating star-rating fas fa-star"></i>${show.rating.average}</span></h2>
+            <p class="card-text"></p> 
+          </div>
+            <img class="fav-shows-img card-img-top card-img-padding col-3" src="${showImg}">
+          
+        </div>
+        `
+    }).join('');
+    // let episodeTrackerList = signedInUser[0].episodeTracker;
+    // let episodeTracker = episodeTrackerList.map((episode) => {
+    //   let episodeImg = show.image?.medium ?? `https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRqEWgS0uxxEYJ0PsOb2OgwyWvC0Gjp8NUdPw&usqp=CAU`;
+    //   return `
+    //       <div class="episode-card card text-center col-2">
+    //         <img class="fav-shows-img card-img-top card-img-padding col-3" src="${showImg}">
+    //         <div class="card-header">
+    //           <h2 class="card-header-text">${show.name}</h2>
+    //           <p class="card-text"><i class="star-rating fas fa-star"></i>${show.rating.average}</p>
+    //         </div>
+    //       </div>
+    //       `
+    // })
+    usersPage.innerHTML = favShowsHtml;
+    similarShows(favShows);
+}
 
-    if (!userDetails.signedIn === true) {
-       console.log(`user is signed in`);
+// similar shows
+function similarShows(showsList) {
+  console.log(showsList);
+  let shows = getAllShows();
+  let getShows = shows.filter((show) => {
+    for (let i = 0; i < showsList.length; i++) {
+      if (show.id == showsList[i]) {
+        return show
+      }
     }
-    
+  });
+  console.log(getShows)
+  for (let i = 0; i < getShows.length; i++) {
+    let showGenres = getShows[i].genres;
+    // let showGenresCheck = showGenres.map((genre) => {
+
+    // });
+    let allSimilarShows = shows.filter((show) => {
+      for (let i = 0; i < showGenres.length; i++) {
+        let genreOne = showGenres[0];
```

Comment:
I don't think it's worth assigning all the array items to their own variables. You'd only really do this if you were accessing the array value a lot of times to save processing time.

For example if you needed to access `arr[1000]` loads of times, it's quicker to set `const item = arr[1000]` so we don't need to do a lookup to item `1000` every time. But `showGeneres` is a very small array and we only access those items once in the code block.